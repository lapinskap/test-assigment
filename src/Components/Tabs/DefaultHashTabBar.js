import PropTypes from 'prop-types';
import React, { useCallback } from 'react';
import { Link } from 'react-router-dom';
import { Card, CardBody } from 'reactstrap';

export default function DefaultHashTabBar({ panels, activeKey, noCards }) {
  const getTabClassesWrapper = useCallback((panel) => {
    const classes = ['rc-tabs-tab'];
    if (activeKey === panel.key) {
      classes.push('rc-tabs-tab-active');
    }
    if (panel.props.disabled) {
      classes.push('rc-tabs-tab-disabled');
    }

    return classes.join(' ');
  }, [activeKey]);

  const hashBar = (
    <div role="tablist" className="rc-tabs-bar" tabIndex="0" data-t1="tabs">
      <div>
        {panels.map((panel) => {
          if (!panel) {
            return false;
          }
          const tab = (
            <div
              data-t1={panel.key}
              role="tab"
              aria-disabled="false"
              aria-selected="true"
              className={getTabClassesWrapper(panel)}
            >
              {panel.props.tab}
            </div>
          );
          if (panel.props.disabled || activeKey === panel.key) {
            return (
              <span key={panel.key} className="rc-tabs-tab-link">
                {tab}
              </span>
            );
          }
          const onClick = panel.onClick || panel.props.onClick;
          return onClick ? (
            <span role="presentation" key={panel.key} className="rc-tabs-tab-link" onClick={onClick}>
              {tab}
            </span>
          ) : (
            <Link key={panel.key} className="rc-tabs-tab-link" to={`#${panel.key}`}>
              {tab}
            </Link>
          );
        })}
      </div>
    </div>
  );
  return (
    noCards ? hashBar
      : (
        <Card className="my-3">
          <CardBody>{hashBar}</CardBody>
        </Card>
      )
  );
}
// No forbidExtraProps because its receive many props from third party library which we do not use
DefaultHashTabBar.propTypes = {
  activeKey: PropTypes.string,
  noCards: PropTypes.bool,
  panels: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    key: PropTypes.string,
    component: PropTypes.node,
    disabled: PropTypes.bool,
    noCards: PropTypes.bool,
    onClick: PropTypes.func,
    props: PropTypes.shape({
      disabled: PropTypes.bool,
      tab: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
    }),
  })),
};
DefaultHashTabBar.defaultProps = {
  activeKey: '',
  panels: [],
  noCards: false,
};
