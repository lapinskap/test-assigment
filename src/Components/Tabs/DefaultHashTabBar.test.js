/* eslint-disable function-call-argument-newline */
import React from 'react';
import renderer from 'react-test-renderer';
import DefaultHashTabBar from './DefaultHashTabBar';
import '../../tests/setupTests';
import TestAppContext from '../../tests/TestAppContext';

describe('<DefaultHashTabBar />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  let panels; let
    activeKey;

  beforeEach(() => {
    panels = [
      {
        props: {
          disabled: true,
        },
      },
      {
        props: {
          display: false,
        },
      },
    ];
    activeKey = 'test';
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default state', () => {
    const tree = renderer
      .create(
        <TestAppContext router>
          <DefaultHashTabBar activeKey={activeKey} panels={panels} />
        </TestAppContext>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
