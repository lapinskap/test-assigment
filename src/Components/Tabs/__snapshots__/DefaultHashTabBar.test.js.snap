// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultHashTabBar /> renders correctly with default state 1`] = `
<div
  className="my-3 card"
>
  <div
    className="card-body"
  >
    <div
      className="rc-tabs-bar"
      data-t1="tabs"
      role="tablist"
      tabIndex="0"
    >
      <div>
        <span
          className="rc-tabs-tab-link"
        >
          <div
            aria-disabled="false"
            aria-selected="true"
            className="rc-tabs-tab rc-tabs-tab-disabled"
            role="tab"
          />
        </span>
        <a
          className="rc-tabs-tab-link"
          href="#/#undefined"
          onClick={[Function]}
        >
          <div
            aria-disabled="false"
            aria-selected="true"
            className="rc-tabs-tab"
            role="tab"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
